### DISCUSSION TITLE
Custom home page

### DISCUSSION THREAD

**Discussion by clementthenard** [outline/outline#4892] [Self Hosting]
Hi, 

I am currently testing your app for my entreprise. It's the perfect tool for us with one exception. I wish I could change the homepage to my liking, but I can't find where I could. Am I stupid?

If i'm not, have you any plan to add it in your roadmap?

Thanks for your amazing work.

**Comment by lampy255** [comment]
Even the ability to assign a specific doc as the homepage would be very useful.

**Comment by laplaceliu** [comment]
It is quite simple to find the location to modify the homepage. You can try to add some html tags in `app/routes/index.tsx`, and then run the program to see what changes have taken place on the homepage.

**Reply by clementthenard** [reply]
Yep, but if we update Outline, then we will lost changes, and I don't want to lost it.

**Comment by Limezy** [comment]
I am using the open source automad CMS (https://automad.org) as a thin front layer on top of Outline, which is installed on a subdomain behind. A few buttons link to either a visitor access or member access.
The design is very close to Outline and both work perfectly together.

Alternatively, you can have one of your document publicly shared, for example `outline.yourinstance.com/s/welcome` with a redirection from `yourinstance.com` to that URL ?

**Comment by laplaceliu** [comment]
As @Limezy said, outline can be regarded as a separate front-end and back-end project. A better approach is to place an API Gateway (such as https://www.krakend.io/ or https://tyk.io/) in front of the outline front end, and configure API forwarding to preserve session, token and other data. You can also replace the Route you want to replace.
